code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -162,7 +170,8 @@ var/global/list/animal_count = list() //Stores types, and amount of animals of t
 
 	if(purge)
 		purge -= 1
-
+	if(isRegenerating)
```

Comment:
If you're gonna reset it like this just remove the if entirely, I checked once and there's no performance improvement.